### PR TITLE
build: integrate AWS SAM

### PR DESCRIPTION
**Pull request by jasonwalsh**
### Purpose

This pull request replaces the workflow of using Terraform to manage lambda deployments with AWS [SAM](https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/what-is-sam.html).

> What is AWS SAM?

AWS [SAM](https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/what-is-sam.html) is an open-source framework that you can use to build serverless applications on AWS.

### Benefits

Unlike Terraform, SAM focuses on managing the lifecycle of lambda functions. At a high level, SAM includes the following:

- Single-deployment configuration
- Local debugging and testing
- Compile dependencies for lambda functions

### Usage

This pull request includes changes to the `requirements.txt` file, so be sure to install the updated dependencies before using SAM:

    $ pip install -r requirements.txt

After installing the required dependencies, invoke the following commands:

    $ sam local start-api > /dev/null 2>&1 &
    [1] 3140
    $ pytest
    $ kill -9 3140

`sam local start-api > /dev/null 2>&1 &` starts SAM locally and runs the process in the background. SAM starts a web server using host `localhost` an port 3000 (e.g., `localhost:3000`).

**Note:** The process will most likely be different, so be sure to remember it. If the process ID is lost, then invoke the following command:

    $ ps aux | grep -v grep | grep start-api

Alternatively, invoke `sam local start-api` in a new terminal window, open a new terminal window and invoke `pytest`.

### TODO

- Create a [`Vagrantfile`](https://www.vagrantup.com/docs/vagrantfile/) with Docker installed to serve as a development environment for hopdrops (see [hopdrops/vagrant](https://github.com/hopdrops/vagrant))
- ~Create the remaining resources in CloudFormation (e.g., [`AWS::Route53::HostedZone`](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-route53-hostedzone.html))~

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -1 +1,5 @@
-squareconnect==2.20190612.1
+aws-sam-cli==0.18.0
+moto==1.3.13
+pytest==5.0.1
+pytest-cov==2.7.1
+squareconnect==2.20190724.0
```
Comment (jasonwalsh) [inline requirements.txt]: This dependency may not be required.

File: template.yaml [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+    Properties:
+      CodeUri: functions/
+      Policies:
+        - AmazonDynamoDBFullAccess
+      Environment:
+        Variables:
+          CLIENT_SECRET: !Ref ClientSecret
+      Handler: redirect.handler
+      Events:
+        oauth:
```
Comment (jasonwalsh) [inline template.yaml]:
```diff
- oauth
+ redirect
```

Comment (jasonwalsh) [reply template.yaml]: See 6021919

File: requirements.txt (hunk 1 above)
Comment (brh55) [reply requirements.txt]: Are we planning on doing a requirements-dev.txt?

**Pull request merged**